classes/local/store/object_file_system.php

```diff
@@ -443,6 +450,14 @@ public function xsendfile($contenthash) {
 
             return $this->redirect_to_presigned_url($contenthash, $headers);
         }
+
+        if ($this->externalclient->support_presigned_urls()) {
+            $filesize = $this->get_filesize_by_contenthash($contenthash);
+            $ranges = $this->get_valid_http_ranges($filesize);
+            if ($filesize > 2097152 && $ranges) {
+                $this->serve_range_request_via_presigned_url($contenthash, $file->get_filesize(), $ranges);
```

Comment:
is this a bug?

                $this->serve_range_request_via_presigned_url($contenthash, $file->get_filesize(), $ranges);

vs

                $this->serve_range_request_via_presigned_url($contenthash, $filesize, $ranges);

Reply:
Yep. Thanks. Fixed